src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,114 @@
+import {
+  ChangeEvent,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  deleteTodo: (id: number) => void,
+  updateTodo: (updatedTodo: Todo) => void,
+};
+
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  updateTodo,
+}) => {
+  const { id, title, completed } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [inputValue, setInputValue] = useState(title);
+
+  const inputEl = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputEl.current) {
+      inputEl.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleTodoDestroy = () => {
+    deleteTodo(id);
+  };
+
+  const handleCompletedTodo = () => {
+    updateTodo({ ...todo, completed: !completed });
+  };
+
+  const handleTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+  };
+
+  const changeTitleBlur = () => {
+    setIsEditing(false);
+    if (inputValue === title) {
+      return;
+    }
+
+    if (inputValue) {
+      updateTodo({ ...todo, title: inputValue });
+    } else {
+      deleteTodo(id);
+    }
```

Comment:
you can use early return 
```suggestion
    if (!inputValue) {
      return deleteTodo(id);
    }
    
    return updateTodo({ ...todo, title: inputValue });
```

Reply:
can't use return for functions, because they return void